nixos/modules/services/misc/gitea.nix

```diff
@@ -203,55 +220,18 @@ in
         };
       };
 
-      appName = mkOption {
-        type = types.str;
-        default = "gitea: Gitea Service";
-        description = lib.mdDoc "Application name.";
-      };
-
       repositoryRoot = mkOption {
         type = types.str;
         default = "${cfg.stateDir}/repositories";
         defaultText = literalExpression ''"''${config.${opt.stateDir}}/repositories"'';
         description = lib.mdDoc "Path to the git repositories.";
       };
 
-      domain = mkOption {
-        type = types.str;
-        default = "localhost";
-        description = lib.mdDoc "Domain name of your server.";
-      };
-
-      rootUrl = mkOption {
-        type = types.str;
-        default = "http://localhost:3000/";
-        description = lib.mdDoc "Full public URL of gitea server.";
-      };
-
-      httpAddress = mkOption {
-        type = types.str;
-        default = "0.0.0.0";
-        description = lib.mdDoc "HTTP listen address.";
-      };
-
-      httpPort = mkOption {
-        type = types.port;
-        default = 3000;
-        description = lib.mdDoc "HTTP listen port.";
-      };
-
       enableUnixSocket = mkOption {
         type = types.bool;
         default = false;
-        description = lib.mdDoc "Configure Gitea to listen on a unix socket instead of the default TCP port.";
-      };
-
-      staticRootPath = mkOption {
-        type = types.either types.str types.path;
-        default = gitea.data;
-        defaultText = literalExpression "package.data";
-        example = "/var/lib/gitea/data";
-        description = lib.mdDoc "Upper level of template and static files path.";
+        visible = false;
+        description = lib.mdDoc ''Exists for backwards compatibility. Use `settings.server.PROTOCOL = "http+unix";'';
```

Comment:
Why not deprecate it then?

Reply:
You can use mkDeprecatedOption or something like this IIRC

Reply:
Found `mkChangedOptionModule` and used that, hopefully that's what @SuperSandro2000 meant.